### PR TITLE
Added the endpoint to get all Shops

### PR DESCRIPTION
**Pull request by Oludayo20**
you can check this one that was used for the previous cohort

# Shops/COM-30

> This could be an existing issue or a linear ticket
>
> - Github Issue Example: My PR Closes Shops
> - Linear Ticket Example: Fixes ID-COM-30

# Changes proposed

I created an endpoint and route that encapsulate the logic for retrieving a list of shops from a database using Prisma. This function handles both success and error scenarios, sending appropriate responses back to the client via the provided Response object. And also wrote the swagger doc.

# Check List (Check all the applicable boxes)

:rotating_light:Please review the [style guide for contributing](add link here) and [guidelines for contributing](add link here) to this repository.

- [V] My code follows the code style of this project.
- [V] This PR does not contain plagiarized content.
- [V] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **main branch** (left side).
- [V] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [V] I am only making changes to files I was requested to.

image

### CODE REVIEW

**Comment by Benrobo** [comment]
fix your merge conflict

**Comment by Oludayo20** [comment]
Done

**Comment by Benrobo** [comment]
resolve conflict please!

**Comment by Oludayo20** [comment]
Done

File: src/controller/shop.controller.ts [hunk 1]
```diff
@@ -71,6 +71,14 @@ export default class ShopController extends BaseController {
     this.success(res, '--shop/deleted', 'shop deleted', 200, null);
   }
 
+  async getAllShops(req: Request, res: Response) {
+    const shops = await prisma.shop.findMany();
```
Comment (Benrobo) [inline src/controller/shop.controller.ts]: this is wrong.. you can't return all shops available on the app.. you only need to return shops created by the user.

File: src/controller/shop.controller.ts [hunk 2]
```diff
@@ -71,6 +71,14 @@ export default class ShopController extends BaseController {
...
+    if (shops.length > 0) {
```
Comment (Benrobo) [inline src/controller/shop.controller.ts]: secondly, you dont need to check if it empty or not, if the shop is empty, an empty array would be returned and the frontend devs can decide to show a message based on that.

File: src/controller/shop.controller.ts [hunk 3]
```diff
@@ -71,6 +71,14 @@ export default class ShopController extends BaseController {
...
+      this.success(res, 'All shops', 'Shops have been listed successfully', 200, shops);
+    } else {
+      this.error(res, '--shops-notfound', 'Shops not found', 404);
```
Comment (Benrobo) [inline src/controller/shop.controller.ts]: sending error isn't ok as this operation isn't an error, it only means the user has no shop, so pls change all this..

**Review by Benrobo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**